src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs

```diff
@@ -61,7 +60,11 @@ string searchToken
 
             try
             {
-                if (_sqlConnection == null || _sqlConnection.State != ConnectionState.Open)
+                // Renew the connection on every call.
+                _sqlConnection = new SqlConnection(_sqlConnString);
```

Comment:
Wrap the _sqlConnection  into Using. It is ok to frequently create/release connection